src/UglyToad.PdfPig/XObjects/XObjectFactory.cs

```diff
@@ -30,15 +29,15 @@ internal static class XObjectFactory
                 throw new InvalidOperationException($"Cannot create an image from an XObject with type: {xObject.Type}.");
             }
 
-            var dictionary = xObject.Stream.StreamDictionary;
+            var dictionary = xObject.Stream.StreamDictionary.Resolve(pdfScanner);
 
             var bounds = xObject.AppliedTransformation.Transform(new PdfRectangle(new PdfPoint(0, 0), new PdfPoint(1, 1)));
 
-            var width = dictionary.Get<NumericToken>(NameToken.Width, pdfScanner).Int;
-            var height = dictionary.Get<NumericToken>(NameToken.Height, pdfScanner).Int;
+            var width = dictionary.Get<NumericToken>(NameToken.Width).Int;
```

Comment:
What's the reason for removing the overload using `pdfScanner` in this file? In general overloads of `Get` and `TryGet` parsing `pdfScanner` or `DirectObjectFinder.Get` are required because most tokens in PDF documents can be indirect references to other tokens, in this case an Indirect reference to a number token in its own object.

Usually where they are used it's because there was already a bug in the code which meant they needed to be added, I just want to check we're not causing a regression in this file 🙏